algebra-core/src/curves/models/short_weierstrass_jacobian.rs

```diff
@@ -603,19 +603,19 @@ impl<P: Parameters> ProjectiveCurve for GroupProjective<P> {
 
         if P::COEFF_A.is_zero() {
             // A = X1^2
-            let a = self.x.square();
+            let mut a = self.x.square();
 
             // B = Y1^2
             let b = self.y.square();
 
             // C = B^2
-            let c = b.square();
+            let mut c = b.square();
 
             // D = 2*((X1+B)2-A-C)
             let d = ((self.x + &b).square() - &a - &c).double();
 
             // E = 3*A
-            let e = a + a.double();
+            let e = a + a.double_in_place();
```

Comment:
Why have we switched to `double_in_place` here?

Reply:
that's reverting vitaly's commit, we've always used double_in_place

Reply:
(see https://github.com/o1-labs/zexe/blob/21a9eb6948083dcad7aa21c61da6feac05e28319/algebra-core/src/curves/models/short_weierstrass_jacobian.rs#L618)